server/src/main/java/com/cloud/storage/snapshot/SnapshotManagerImpl.java

```diff
@@ -1621,7 +1621,8 @@ public SnapshotInfo takeSnapshot(VolumeInfo volume) throws ResourceAllocationExc
             if (snapshotStrategy == null) {
                 _snapshotDao.remove(snapshotId);
                 logger.debug("No strategy found for creation of snapshot [{}], removing its record from the database.", snapshot);
-                throw new CloudRuntimeException(String.format("Can't find snapshot strategy to deal with snapshot:%s", snapshot.getSnapshotVO()));
+                throw new UnsupportedOperationException(String.format("Unable to find a snapshot strategy to create snapshot [%s] of volume [%s]. Please check the logs.",
```

Comment:
@JoaoJandre 
thanks for the PR

this is not user-friendly I think, only administrator can access the check the logs

Reply:
@weizhouapache How would you change the message so that it is still generic (this is an error when no snapshot strategy is found; thus this message will be shown for other situations as well) and does not reveal the infrastructure information to the end user?